v0/destinations/mautic/transform.js

```diff
@@ -0,0 +1,247 @@
+    response.body.FORM = removeUndefinedAndNullValues(payload);
+    response.endpoint = endpoint;
+    const authKey = await getEncodedAuth(destination);
+    response.headers = {
+      Authorization: `Basic ${authKey}`
+    };
+    response.method = method;
+    return response;
+  }
+  throw new CustomError("Payload could not be Constructed");
+};
+const responseBuilderGroup = async (endpoint, destination) => {
+  const response = defaultRequestConfig();
+  response.endpoint = endpoint;
+  const authKey = await getEncodedAuth(destination);
+  response.headers = {
+    Authorization: `Basic ${authKey}`
+  };
+  response.method = "POST";
+  return response;
+};
+
+let BASE_URL;
+
+// for validating email match function outdated
+function ValidateEmail(inputText) {
+  console.log(inputText);
+  return true;
+  // const mailformat = new RegExp"/^w+([.-]?w+)*@w+([.-]?w+)*(.w{2,3})+$/";
+  // // console.log(inputText.value);
+  // if (inputText.value.Match(mailformat)) {
+  //   return true;
+  // }
+  // return false;
+}
+// for validating Phone match function outdated
+function validatePhone(inputText) {
+  const phoneno = /^\d{10}$/;
+  if (inputText.match(phoneno)) {
+    return true;
+  }
+  return false;
+}
+
+const groupResponseBuilder = (message, destination) => {
+  console.log("Inside");
+  let groupClass;
+  switch (message.traits.type.toLowerCase()) {
+    case "segments":
+      groupClass = "segemnts";
+      break;
+    case "campaigns":
+      groupClass = "campaigns";
+      break;
+    case "companies":
+      groupClass = "companies";
+      break;
+    default:
+      throw new CustomError("This grouping is npt supported");
+  }
+  const endpoint = `${BASE_URL}/${groupClass}/${message.groupId}/contact/${message.context.externalId}/add`;
+  return responseBuilderGroup(endpoint, destination);
+};
+const identifyResponseBuilder = (message, destination) => {
+  const { Config } = destination;
+  // checking for message details validations
+  if (
+    message.context.traits.email &&
+    !ValidateEmail(message.context.traits.email)
+  ) {
+    throw CustomError("Invalid Mail Provided", 400);
+  }
+  if (
+    message.context.traits.Phone &&
+    !validatePhone(message.context.traits.Phone)
+  ) {
+    throw CustomError("Invalid Phone No. Provided", 400);
+  }
+  if (message.context.traits.title && !titles.includes(message.title)) {
+    throw CustomError("This title is not supported", 400);
+  }
+  if (message.originalTimestamp && !isDate(message.originalTimestamp)) {
+    throw CustomError("Date is Invalid", 400);
+  }
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.IDENTIFY.name]
+  );
+  if (message.traits.address || message.context.traits.address) {
+    let add;
+    if (message.traits.address) {
+      add = message.traits.address;
+    } else {
+      add = message.context.traits.address;
+    }
+    const len = add.length;
+    if (len > 128) {
```

Comment:
We do not need to throw error for this. Because this is a non-mandatory field. Rather will mention in the documentation that address will be trimmed upto the first 128 characters.